docs/plugins/contributors/gatsby-node.js

```diff
@@ -0,0 +1,56 @@
+const { Octokit } = require("@octokit/rest");
+
+const fallback = require("./fallback.json");
+
+const NODE = `contributor`;
+
+exports.sourceNodes = async ({ actions, createContentDigest, createNodeId }, { repo, owner }) => {
+  const { createNode } = actions;
+
+  try {
+    const octokit = new Octokit({ auth: process.env.GH_TOKEN });
+    const names = [];
+
+    await octokit.repos
+      .listContributors({
+        owner,
+        repo,
+      })
+      .then(c => names.push(...c.data.map(({ login }) => login)));
+
+    const reqs = names.map(username => octokit.users.getByUsername({ username }));
+    const users = await Promise.all(reqs).then(all => all.map(n => n.data));
+
+    users.forEach(u => {
+      createNode({
+        ...u,
+        id: createNodeId(`${NODE}-${u.id}`),
+        parent: null,
+        children: [],
+        internal: {
+          type: NODE,
+          content: JSON.stringify(u),
+          contentDigest: createContentDigest(u),
+        },
+      });
+    });
+  } catch (error) {
+    if (error.status === 401) {
+      console.error("401: UNAUTHORIZED, CREATE GITHUB TOKEN");
+      createNode({
+        ...fallback,
+        id: createNodeId(`${NODE}-${fallback.id}`),
+        parent: null,
+        children: [],
+        internal: {
+          type: NODE,
+          content: JSON.stringify(fallback),
+          contentDigest: createContentDigest(fallback),
+        },
+      });
+    } else {
+      console.error(error);
+    }
+  }
+  return undefined;
+};
```

Comment:
Instead of all of this this custom plugin could serve for providing an empty array as the GraphQL response that `gatsby-source-graphql` provides. Then in `gatsby-config.js` you'd check whether `GH_TOKEN` exists, if it does configure `gatsby-source-graphql`, if it doesn't configure the custom plugin. (Then in the component you'd check whether the array is empty etc. etc.)

What do you think?

Reply:
i don't think we should have them both, custom plugin or `gatsby-source-graphql`, it's same logic twice, custom plugin can be run without providing token at all, that is the difference with `gatsby-source-graphql`, which needs that token in `gatsby-config.js`. 🤔